src/main/java/io/debezium/connector/cassandra/CassandraConnectorContext.java

```diff
@@ -23,13 +24,18 @@
     private final ChangeEventQueue<Event> queue;
     private final SchemaHolder schemaHolder;
     private final OffsetWriter offsetWriter;
+    private final HashSet<String> errorCommitLogSet;
```

Comment:
Let's just expose `Set` here and as the return value of the method (here and elsewhere in this change set). As a name perhaps just `erroneousCommitLogs`, or similar? No need to encode the type into the variable name.

Reply:
Is this set accessed from multiple threads at the same time btw.?

Reply:
> Let's just expose `Set` here and as the return value of the method (here and elsewhere in this change set). As a name perhaps just `erroneousCommitLogs`, or similar? No need to encode the type into the variable name.

Thank you for the suggestion! Updated accordingly.

Reply:
> Is this set accessed from multiple threads at the same time btw.?

Nice catch! There are multiple threads that can access this set:
- But read access (`set.contains(commitLog-1)`) will always happen after write access (`set.add(commitLog-1)`). 
- Write access (`set.contains(commitLog-1)` and `set.contains(commitLog-2)`) from different threads however can happen at the same time.
I'm not sure if we need to be concerned about thread safe in this case ^ . Do you suggest any safer kind of Set to use please?

Reply:
Assuming there won't be high contention here (i.e. not many concurrent requests at the same time, which I think is not the case, as log files aren't added like every second or so), the simplest is make all methods accessing it `synchronized`. That way, only one thread at a time will access the set.